src/Rules.php

```diff
@@ -31,13 +31,17 @@ public function add($rule): void
         $this->rules[] = $this->normalizeRule($rule);
     }
 
-    public function validate($value, DataSetInterface $dataSet = null, bool $previousRulesErrored = false): Result
+    public function validate($value, ValidationContext $context = null): Result
     {
+        $context = $context ?? new ValidationContext();
```

Comment:
Shouldn't it be placed into `Validator`?

Reply:
No.

`Validator` and separate rules can be use without context. But `Rules::validate` use context for check previous rules errored, because we create context here.